processing/src/main/java/org/apache/druid/segment/Metadata.java

```diff
@@ -233,6 +260,39 @@ public String toString()
            ", timestampSpec=" + timestampSpec +
            ", queryGranularity=" + queryGranularity +
            ", rollup=" + rollup +
+           ", sortOrder=" + sortOrder +
            '}';
   }
+
+  @Nullable
+  private static List<String> mergeSortOrders(List<List<String>> sortOrdersToMerge)
+  {
+    if (sortOrdersToMerge.isEmpty()) {
+      return null;
+    }
+
+    if (sortOrdersToMerge.stream().anyMatch(Objects::isNull)) {
+      return null;
```

Comment:
should this be an error if they aren't all null? seems like it should be pretty consistent across indexable adapters...

Reply:
I made it lenient for two reasons:

- Merging of the other parts of the Metadata is also lenient
- It isn't documented that this method requires that all Metadatas are sourced from segments created with the same ingestion spec + the same version of the software